### ISSUE TITLE
[UNT0022] SetPositionAndRotation diagnostic is reporting wrong positives

### ISSUE THREAD

**Issue by sailro** [microsoft/Microsoft.Unity.Analyzers#145]
#### To Reproduce
```csharp
        GameObject instance = ...;
        GameObject go = ...;
        instance.transform.position = go.transform.position;
        go.transform.rotation = go.transform.rotation;
```
#### Expected behavior
Nothing should be reported, because we are not working on the same transform.

#### Actual behavior
Reported with the following codefix:

```csharp
        GameObject instance = ...;
        GameObject go = ...;
        transform.SetPositionAndRotation(go.transform.position, go.transform.rotation);
```
